src/enterdb.erl

```diff
@@ -529,6 +530,55 @@ read_range_n_ts_(Tab, TD, {ok, DBKey, HashKey}, N, false) ->
 read_range_n_ts_(_Tab, _TD, {error, _} = E, _N, _) ->
     E.
 
+%%--------------------------------------------------------------------
+%% @doc
+%% Reads N number of Keys from table time_series table
+%% with name Name starting form StartKey.
+%% Stop if we hit StopKey before N number of results are returned.
+%% @end
+%%--------------------------------------------------------------------
+-spec read_range_n_ts(Name :: string(),
+		      StartKey :: key(),
+		      StopTs  :: {string(), term()},
+		   N :: pos_integer()) ->
+    {ok, [kvp()]} | {error, Reason :: term()}.
+read_range_n_ts(Name, StartKey, StopTs, N) ->
+    case enterdb_lib:get_tab_def(Name) of
+	TD = #{distributed := Dist}->
+	    DBKey = enterdb_lib:make_key(TD, StartKey),
+	    SKey = lists:keyreplace("ts", 1, StartKey, StopTs),
+	    SDBKey = enterdb_lib:make_key(TD, SKey),
+	    read_range_stop_n_ts_(Name, TD, DBKey, SDBKey, N, Dist);
+	{error, _} = R ->
+	    R
+    end.
+
+-spec read_range_stop_n_ts_(
+	Name :: string(),
+	TD :: #{},
+	{ok, DBKey :: binary(), binary()},
+	{ok, SDBKey :: binary(), binary()},
+	N :: pos_integer(),
+	Dist :: true | false) ->
+    {ok, [kvp()]} | {error, Reason :: term()}.
+read_range_stop_n_ts_(Tab, TD,
+		{ok, DBKey, HashKey},
+		{ok, SDBKey, _},
+		N, true) ->
+    {ok, {Shard, Ring}} = gb_hash:get_node(Tab, HashKey),
+    ?dyno:call(Ring,
+	       {enterdb_lib, read_range_n_on_shard_ts,
+		[Shard, TD, HashKey, DBKey, SDBKey, N]}, read_range_stop_n_ts_);
+read_range_stop_n_ts_(Tab, TD,
+		{ok, DBKey, HashKey},
+		{ok, SDBKey, _}, N, false) ->
+    {ok, Shard} = gb_hash:get_local_node(Tab, HashKey),
+    enterdb_lib:read_range_n_on_shard_ts(Shard, TD, HashKey, DBKey, SDBKey, N);
+read_range_stop_n_ts_(_Tab, _TD, {error, _} = E, _, _N, _) ->
+    E;
+read_range_stop_n_ts_(_Tab, _TD, _, {error, _} = E, _N, _) ->
+    E.
+
```

Comment:
Why do we need this when there is read_range/3? Or can't we add a new function called  read_range_ts/3?